### ISSUE TITLE
Add quick setting to video panel for "GIF-like" playback

### ISSUE THREAD

**Issue by pbakaus** [WordPress/gutenberg#19730]
**Is your feature request related to a problem? Please describe.**
GIFs must die as format, as they have file size, color palette and performance issues. Much as been written about it, the best article is [probably this WebFundamentals article](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/replace-animated-gifs-with-video).

**Describe the solution you'd like**
It would be excellent if Gutenberg could steer people into using videos by making "gif-like" playback just one click away (most journalists likely don't know which options to toggle). Here's a concept mock (but this certainly doesn't have to be the exact solution):
![animation_gutenberg](https://user-images.githubusercontent.com/43004/72634674-e192ac00-390f-11ea-8116-7f9958bab8be.png)

I [blogged about it here](https://paulbakaus.com/2020/01/17/gifs-must-die/) as well (calling out Gutenberg, but mostly as positive example, not to single it out).

**Describe alternatives you've considered**
An alternative that should probably also be done but might warrant a separate issue is the auto-conversion of GIFs you drag into Gutenberg to mp4/webm combinations. Platforms like Twitter already do this, and WordPress/Gutenberg would be doing a great service to the planet by wasting less CPU cycles for billions of people.

/cc @westonruter

**Comment by westonruter** [comment]
I don't think it is feasible to transcode GIF to MP4 on the server given the wide variety of capabilities among hosts. What would be interesting is doing this conversion client-side with Wasm. If someone drags a GIF into the editor or adds a GIF to the media library, before uploading it could:

1. Transcode from GIF to MP4.
2. Generate poster image frame for the video to set as the attachment's featured image.

Disclaimer: I'm not a Wasm expert, or even a novice.

**Comment by pbakaus** [comment]
> 1. Transcode from GIF to MP4.
> 2. Generate poster image frame for the video to set as the attachment's featured image.
> 
> Disclaimer: I'm not a Wasm expert, or even a novice.

Totally possible. Promising new actively maintained ffmpeg.js: https://github.com/ffmpegjs/ffmpeg.js

**Comment by swissspidy** [comment]
Very promising stuff!

I think client-side asset optimization and transcoding definitely warrants its own issue and perhaps even a [Trac ticket](https://core.trac.wordpress.org/) (as it doesn't affect only Gutenberg).

In the meantime, such a UI change seems like a great first step. There might also be other touchpoints for this. For example, when a user uploads a gif, the editor could suggest using a video instead.

**Comment by swissspidy** [comment]
I've been solving this using the variations feature. It works pretty smooth:

https://github.com/user-attachments/assets/ecf0d904-6619-4e73-8573-866e0f445313

I'm going to demo this at WCUS this month, and will continue working on it as part of  #61447

**Comment by karmatosed** [comment]
I'm not quite sure what we are asking for with the needs design label so for now going to just add the feedback label. Once we work out the exact requirements then we absolutely can have some explorations around this. I am though a little unsure what the decision is right now to then have it go straight to design.

**Comment by jasmussen** [comment]
This was very recently resolved. The Video block now has a built-in GIF variation (packages/block-library/src/video/variations.js) that plays back as a muted, looping, autoplaying, controls-free video, switchable in one click via the block's "Display as" toolbar, and animated GIFs uploaded to the editor are automatically transcoded to MP4/WebM client-side via #78410 ("Animated GIF to video conversion (via mediabunny) plus conversion controls," merged). Both alternatives requested in this issue are shipped.
